app/src/main/java/com/github/epfl/meili/forum/Post.kt

```diff
@@ -0,0 +1,36 @@
+package com.github.epfl.meili.forum
+
+import android.os.Parcelable
+import android.util.Log
+import com.google.firebase.firestore.DocumentSnapshot
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+data class Post(
+    val id: String,
+    val author: String,
+    val title: String,
+    val text: String
+) : Parcelable {
+
+    /** Creates an instances of Post from other objects */
+    companion object {
+        private const val TAG = "Post"
+
+        // Create Post from Firestore data
+        fun DocumentSnapshot.toPost(): Post? {
+            return try {
+                val id = id
+                val author = getString("username")!! // !! Means we guarantee not null
+                val title = getString("title")!!
+                val text = getString("text")!!
+                Post(id, author, title, text)
+            } catch (e: Exception) {
+                Log.e(TAG, "Error converting to Post")
+                null // Return null if error occurs
+            }
```

Comment:
I think we can remove this method entirely by making the data class's fields vars and assigning them default values, as done here:

https://github.com/meili-epfl/Meili/blob/main/app/src/main/java/com/github/epfl/meili/models/Review.kt

Reply:
This is used to convert the database information to a Post. So it is needed for when I retrieve information from Firestore no ?

Reply:
It is already implemented actually, you can call `documentSnapshot.toObject(Post::class.java)`